src/main/kotlin/build/buf/gradle/AbstractBufExecTask.kt

```diff
@@ -0,0 +1,130 @@
+// Copyright 2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.gradle
+
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+/**
+ * A task executing buf executable as part of its operation.
+ */
+abstract class AbstractBufExecTask : AbstractBufTask() {
+    /** The buf executable. */
+    @get:InputFiles
+    abstract val bufExecutable: ConfigurableFileCollection
```

Comment:
Just noticed this - what makes this `InputFiles` and a `ConfigurableFileCollection` instead of `InputFile` and a `Property<File>`?

Reply:
I used `ConfigurableFileCollection` to make it easily set from "gradle configuration" (see the setting of this property).

Reply:
It saves the call to `singleFile` - is that right? 

`bufExecutable.set(configurations.getByName(BUF_BINARY_CONFIGURATION_NAME).singleFile`

But you end up having to call that later. Perhaps having a `Property<File>` is more clear.

Reply:
If I remember correctly, I cannot just call `singleFile` at that point, because the dependency is not yet resolved. So later it is resolved, the better.

Reply:
Ah, makes sense.